pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/controller/periodic/tasks/SegmentStatusCheckerIntegrationTest.java

```diff
@@ -243,6 +243,16 @@ public void testSegmentStatusChecker() {
     Assert.assertEquals(
         controllerMetrics.getValueOfTableGauge(errorOfflineTable, ControllerGauge.PERCENT_SEGMENTS_AVAILABLE), 100);
 
+    // additional 30 seconds wait, to ensure realtime table setup has also completed
```

Comment:
Can this be made conditional? How do we know 30s is enough?

Reply:
By conditional, do you mean not have the 30 seconds timeout at all? Or take a different path and not do the tests if 30s is up?